src/outgoing.rs

```diff
@@ -69,6 +80,24 @@ impl FromStr for Outgoing {
   }
 }
 
+impl Serialize for Outgoing {
+  fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+  where
+    S: Serializer,
+  {
+    serializer.collect_str(self)
+  }
+}
+
+impl<'de> Deserialize<'de> for Outgoing {
+  fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+  where
+    D: Deserializer<'de>,
+  {
+    Ok(DeserializeFromStr::deserialize(deserializer)?.0)
+  }
+}
+
 #[cfg(test)]
```

Comment:
We should have a test for display, and tests for serialize/deserialize that round trip one of each variant.